### ISSUE TITLE
Move AWS config out of AWS UI and into config files

### ISSUE THREAD

**Issue by wholesomedev** [hollowverse-archive/hollowverse#495]
Related issue #233 

Related discussion on Discord:

```
[12:39 PM] mk: @fwz if we move Route 53 and the DB to Serverless, will we still have to tweak their configurations manually on the AWS UI for any reason?
[12:41 PM] fwz: @mk Based on on my observation, it would be possible, but any updates to the CloudFormation template could override the changes. Although it seems that if a particular setting is not specified in the CloudFormation template, it may be kept as-is when the stack is updated.(edited)
[12:43 PM] fwz: DBs can be moved quite easily, but I'm not sure about the Route 53 hosted zones (there does not seem to be a reliable plugin to set the records, and it would be tricky to reference one stack's endpoints in another stack)
[12:46 PM] mk: @fwz So our goal is to have our entire infrastructure as text files, right?

I'm wondering if it makes sense to treat Serverless-generated CloudFormations as artifacts which we shouldn't modify. And then move the rest of our infrastructure configurations to version-controlled textfiles on GitHub that we deploy to AWS. The CloudFormation page seems to indicate that that's possible? https://aws.amazon.com/cloudformation/
[12:47 PM] mk: That way we would be 100% out of the AWS UI
[12:50 PM] fwz: @mk I see your point. We could create a dedicated CloudFormation stack for the databases and the other stacks managed by Serverless could continue to work without any changes. However, there will still be some manual configuration required. For example, when we define a database in the CloudFormation template (which is either a JSON or a YAML file), the created database will have a randomly generated subdomain which we will need to use to connect to the database in the API stack. We cannot know in advance what the subdomain will be so we will need to create the stack, get the connection URL, and hardcode it into the API stack.
[12:52 PM] fwz: Same thing goes for CloudFront distributions and Route 53 records. We would need to create the CloudFront distro and then get the AWS domain name for that distro (*.cloudfront.net) in order to alias a subdomain to that distro (api.hollowverse.com => *.cloudfront.net).
[12:54 PM] fwz: The nice thing about Serverless and its plugins is that it can automate this provided that all the resources are defined in the same stack. If we define an API Gateway and a CloudFront distro and a subdomain in the same Serverless config, we can link all these resources together without having to know what the endpoint URLs are for each of these resources.
[12:55 PM] fwz: So what I was thinking is that if we manage to put every resource that we currently manage manually in one of the Serverless stacks that we have, we would not have to hardcode things.
[12:59 PM] mk: @fwz Sure. If putting everything in Serverless is possible, that would make things more consistent for us.

In case some things prove hard to move to Serverless, I'm wondering if CloudFormation config files already provide means for referencing generated resources with variable names...perhaps by...creating variable names, like we do in programming, in the config files :smiley:(edited)
[1:00 PM] mk: Some testimonials in the CloudFormation page reference having one-click deploy of their entire infrastructure
[1:05 PM] fwz: CloudFormation templates have a field called "Outputs" which contains things that are generated when the stack is created (e.g. connection URL for a database, API URL for API Gateway, domain name for CloudFront distro). I'm just wondering how we could reference these in another CloudFormation stack. Another thing that needs to be figured out is: how would be able to define the passwords for the database in the template and update the secret in Secrets Manager so the API could connect to the database.

Lots of things to figure out but I think all of that can be done if we want to put work into it.
[1:08 PM] mk: @fwz I think it's worth it to move out of AWS UI and into config files. I can create a feature blocker issue for that, and you can get to looking into it when you get a chance?
```

**Comment by forabi** [comment]
This can be closed once we move the API to use the new infrastructure.

**Comment by wholesomedev** [comment]
What about the automatic infrastructure deployment? Should we track that in another issue?

**Comment by forabi** [comment]
Yeah I think so. I'm going to open a new issue for that.

**Comment by forabi** [comment]
Oops! This should not have been closed by hollowverse/architecture#4. Anyway, both this and #233 can now be closed.
